spinalcordtoolbox/deepgmseg/deepgmseg.py

```diff
@@ -0,0 +1,215 @@
+    """
+    y, x = img.shape
+
+    startx = x // 2 - (cropx // 2)
+    starty = y // 2 - (cropy // 2)
+
+    if startx < 0 or starty < 0:
+        raise RuntimeError("Negative crop.")
+
+    cropped_region = CroppedRegion((x, y), (startx, starty),
+                                   (cropx, cropy))
+
+    return img[starty:starty + cropy,
+               startx:startx + cropx], cropped_region
+
+
+def threshold_predictions(predictions, thr=0.999):
+    """This method will threshold predictions.
+
+    :param thr: the threshold.
+    :return: thresholded predictions.
+    """
+    thresholded_preds = predictions[:]
+    low_values_indices = thresholded_preds <= thr
+    thresholded_preds[low_values_indices] = 0
+    low_values_indices = thresholded_preds > thr
+    thresholded_preds[low_values_indices] = 1
+    return thresholded_preds
+
+
+def segment_volume(ninput_volume, model_name):
+    """Segment a nifti volume.
+
+    :param ninput_volume: the input volume.
+    :param model_name: the name of the model to use.
+    :return: segmented slices.
+    """
+    gmseg_model_challenge = DataResource('deepgmseg_models')
+    model_path, metadata_path = model.MODELS[model_name]
+
+    metadata_abs_path = gmseg_model_challenge.get_file_path(metadata_path)
+    with open(metadata_abs_path) as fp:
+        metadata = json.load(fp)
+
+    deepgmseg_model = model.create_model(metadata['filters'])
+
+    model_abs_path = gmseg_model_challenge.get_file_path(model_path)
+    deepgmseg_model.load_weights(model_abs_path)
+
+    volume_data = ninput_volume.get_data()
+    axial_slices = []
+    crops = []
+
+    for slice_num in xrange(volume_data.shape[2]):
+        data = volume_data[..., slice_num]
+        data, cropreg = crop_center(data, model.CROP_HEIGHT,
+                                    model.CROP_WIDTH)
+        axial_slices.append(data)
+        crops.append(cropreg)
+
+    axial_slices = np.asarray(axial_slices, dtype=np.float32)
+    axial_slices = np.expand_dims(axial_slices, axis=3)
+
+    axial_slices -= metadata['mean_train']
+    axial_slices /= metadata['std_train']
+
+    preds = deepgmseg_model.predict(axial_slices, batch_size=8,
+                                    verbose=True)
+    preds = threshold_predictions(preds)
+    pred_slices = []
+
+    # Un-cropping
+    for slice_num in xrange(preds.shape[0]):
+        pred_slice = preds[slice_num][..., 0]
+        pred_slice = crops[slice_num].pad(pred_slice)
+        pred_slices.append(pred_slice)
+
+    pred_slices = np.asarray(pred_slices, dtype=np.uint8)
+    pred_slices = np.transpose(pred_slices, (1, 2, 0))
+
+    return pred_slices
+
+
+def segment_file(input_filename, output_filename,
+                 model_name, verbosity):
+    """Segment a volume file.
+
+    :param input_filename: the input filename.
+    :param output_filename: the output filename.
+    :param model_name: the name of model to use.
+    :param verbosity: the verbosity level.
```

Comment:
`:return: output file name` (but why?)

Reply:
WIll add the doc. Let's say that in future we change the output file name or output path, if we enforce this return since now we'll have less work in future to change everyone that is calling this function. If I return None I'll be losing this advantage.